app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -768,19 +774,26 @@ public function getBoltOrderToken($quote, $checkoutType)
      * @return string javascript Promise string used for initializing BoltCheckout
      */
     public function getBoltOrderTokenPromise($checkoutType) {
-
+        $onError = /** @lang JavaScript */<<<JS
+if (typeof BoltPopup !== 'undefined' && typeof response.responseJSON.error_messages === 'string') {
+    BoltPopup.setMessage(response.responseJSON.error_messages);
+    BoltPopup.addOnCloseCallback(function() {location.reload();});
+    BoltPopup.show();
+}
+JS;
+        $checkoutTokenUrl = $this->boltHelper()->getMagentoUrl("boltpay/order/create/checkoutType/$checkoutType");
+        $parameters = "''";
         if ( $checkoutType === Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_FIRECHECKOUT ) {
             $checkoutTokenUrl = $this->boltHelper()->getMagentoUrl('boltpay/order/firecheckoutcreate');
             $parameters = 'checkout.getFormData ? checkout.getFormData() : Form.serialize(checkout.form, true)';
         } else if ( $checkoutType === Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_ADMIN ) {
             $checkoutTokenUrl = $this->boltHelper()->getMagentoUrl("adminhtml/sales_order_create/create/checkoutType/$checkoutType", array(), true);
             $parameters = "''";
-        } else {
-            $checkoutTokenUrl = $this->boltHelper()->getMagentoUrl("boltpay/order/create/checkoutType/$checkoutType");
-            $parameters = "''";
+        } else if($checkoutType === Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_MULTI_PAGE){
```

Comment:
What about the else case?

Reply:
@adentong 
The else statement was moved to the top above the conditional actions.  Perhaps a switch statement would be more elegant. I can change that to make it more clear.

Reply:
Ah yes I see. Yea switch would probably be a bit more clear.